### PR TITLE
[WIP] Update status document with test coverage information

### PR DESCRIPTION
**Pull request by itaiferber**
This brings [`Docs/Status.md`](/apple/swift-corelibs-foundation/blob/master/Docs/Status.md) up to date with the current state of implementation as part of an effort to improve documentation and test coverage across the project. With up-to-date information, it will be easier to see what components of our project require additional work, and more importantly, where we can improve in testing.

This PR is not ready to merge as I will be pushing additional commits showing the actual state of test coverage, but I wanted to get this up in case anyone had comments or concerns with this initiative. I also intend to update our contribution guidelines to highlight that, ideally, PRs and changes to our codebase should also come with update to sections in the corresponding tables to reflect coverage and completeness.

(I recommend [viewing the document](https://github.com/apple/swift-corelibs-foundation/blob/ac40293f61a53c8e1cdc768c03dc8aa3c8a24dd4/Docs/Status.md) to see what the proposed table format looks like.)

### CODE REVIEW

**Comment by parkera** [comment]
Looks great!

**Comment by itaiferber** [comment]
@parkera Thanks. :)
I originally had an additional column in each table for what file every entity was defined in (for easy lookup, and to note that, say, `NSDateInterval` is actually defined in `NSDate.swift`), but the tables are already pretty wide... Not sure if that's information we care to preserve, but it's an option.

In any case, I'll be updating this over the course of the next day or so so we hopefully get a better idea of where we're really lacking.

File: Docs/Status.md [hunk 1]
```diff
@@ -7,151 +7,301 @@ Foundation is divided into groups of related functionality. These groups are cur
...
+    |------------------|-----------------|---------------|-------------------------------------------------------------------------------|
+    | `Port`           | Unimplemented   | ?             |                                                                               |
+    | `MessagePort`    | Unimplemented   | ?             |                                                                               |
+    | `SocketPort`     | Unimplemented   | ?             |                                                                               |
+    | `PortMessage`    | Unimplemented   | ?             |                                                                               |
+    | `RunLoop`        | Mostly Complete | ?             | `add(_: Port, forMode:)` and `remove(_: Port, forMode:)` remain unimplemented |
+    | `NSStream`       | Mostly Complete | ?             |                                                                               |
+    | `Stream`         | Unimplemented   | ?             | Methods which do not require a concrete implementation remain unimplemented   |
```
Comment (iainsmith) [inline Docs/Status.md]: This sentence left me a bit confused. I was wondering if you meant one of the following?

> Methods which ~~do not~~ require a concrete implementation remain unimplemented

or

> Methods which do not require a concrete implementation need updating to reflect this.

Comment (itaiferber) [inline Docs/Status.md]: Good eye, thanks for catching this. I'd meant the first one — all methods which are not implemented in terms of `NSRequiresConcreteImplementation()` have been left `NSUnimplemented()`, and need implementations. I will push a correction.

**Comment by itaiferber** [comment]
@parkera When you get the chance, can you please review the changes I've added to our guidelines? I want to make sure the public-facing text is appropriate

**Comment by itaiferber** [comment]
@parkera @phausler @michael-lehew Also, if you guys wouldn't mind sanity-checking the [latest revision](https://github.com/apple/swift-corelibs-foundation/blob/d98174633d7f596c991861f0060e868d4db38e5d/Docs/Status.md), I would appreciate it

I want to make sure that everything seems reasonable before merging

**Comment by parkera** [comment]
LGTM

**Pull request merged**